crates/biome_js_analyze/tests/specs/nursery/useNumberProperties/invalid.js

```diff
@@ -0,0 +1,80 @@
+const foo = { [NaN]: 1 };
+
+const foo = { [NaN]() {} };
+
+foo[NaN] = 1;
+
+class A {
+	[NaN]() {}
+}
+
+foo = { [NaN]: 1 };
+
+const foo = Infinity;
+
+if (Object.is(foo, Infinity)) {
+}
+
+const foo = bar[Infinity];
+
+const foo = { Infinity };
+
+const foo = { Infinity: Infinity };
+
+const foo = { [Infinity]: -Infinity };
+
+const foo = { [-Infinity]: Infinity };
+
+const foo = { Infinity: -Infinity };
+
+const { foo = Infinity } = {};
+
+const { foo = -Infinity } = {};
+
+const foo = Infinity.toString();
+
+const foo = -Infinity.toString();
+
+const foo = (-Infinity).toString();
+
+const foo = +Infinity;
+
+const foo = +-Infinity;
+
+const foo = -Infinity;
+
+const foo = -(-Infinity);
+
+const foo = 1 - Infinity;
+
+const foo = 1 - -Infinity;
+
+const isPositiveZero = (value) => value === 0 && 1 / value === Infinity;
+
+const isNegativeZero = (value) => value === 0 && 1 / value === -Infinity;
+
+const { a = NaN } = {};
+
+const { [NaN]: a = NaN } = {};
+
+const [a = NaN] = [];
+
+function foo({ a = NaN }) {}
+
+function foo({ [NaN]: a = NaN }) {}
+
+function foo([a = NaN]) {}
+
+function foo() {
+	return -Infinity;
+}
+
+globalThis.parseFloat(foo);
+
+global.parseFloat(foo);
```

Comment:
`global_identifier` doesn't support `global`. `global` was a candidate name for global name spacing before the standardization of `globalThis`.

```suggestion
// Biome doesn't suport `global` for global namespacing
// global.parseFloat(foo);
```

Reply:
Thanks for review!

> By the way, why did you prefer useNumberProperties over useNumberNamespace?

I hadn't thought deeply about it, but since the term `properties` was used in eslint ([prefer-number-properties](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/prefer-number-properties.md)), I decided to go along with that.

fix: 99e0270